charms/worker/k8s/src/charm.py

```diff
@@ -567,6 +595,10 @@ def _update_status(self):
         if version := self._get_snap_version():
             self.unit.set_workload_version(version)
 
+        if not self._is_node_ready():
+            status.add(ops.WaitingStatus("Node not yet Ready"))
+            return
```

Comment:
This is actually a nice little check for update-status to keep the node from showing "Ready" when -- actually it can't host any workloads yet because it's not Kubernetes Ready